RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -500,9 +707,134 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
 	      }
           } // end of HBHE check
      } // loop on HcalCalibDigiCollection
+
+     // now match the laser monitor data by fiber (in time) 
+     if( fillLaserMonitor_ ) {
+       // check for any fibers without data and fill
+       // them so we dont run into problems later
+       for( unsigned idx = 0; idx < laserMonCBoxList_.size(); ++idx ) {
+           if( lasmon_adcs[idx].empty() ) {
+               lasmon_adcs[idx] = std::vector<int>(10, -1);
+           }
+           if( lasmon_capids[idx].empty() ) {
+               lasmon_capids[idx] = std::vector<int>(10, -1);
+           }
+       }
+       unsigned nFibers = laserMonIEtaList_.size();
+       // for each fiber we need to find the index at with the 
+       // data from the next fiber matches in order to stitch them together.
+       // When there is an overlap, the data from the end of the
+       // earlier fiber is removed.  There is no removal of the last fiber
+       std::vector<unsigned> matching_idx; 
+       // we assume that the list of fibers was given in time order
+       // (if this was not the case, then we just end up using 
+       // all data from all fibers )
+       for( unsigned fidx = 0; fidx < (nFibers - 1); ++fidx ) {
```

Comment:
This fails if nFibers=0, which is not allowed by construction, but it is anyhow worth putting a protection:
```
       for( unsigned fidx = 0; nFibers>0 && fidx < (nFibers - 1); ++fidx ) {
```

Reply:
OK